File: src/test/java/com/claymccoy/jcalc/CalculatorImplTest.java
package com.claymccoy.jcalc;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

import com.claymccoy.jcalc.model.Expression;
import com.claymccoy.jcalc.model.Operand;
import com.claymccoy.jcalc.model.Operator;

public class CalculatorImplTest {
    
    @Test
    public void testAddition() {
        CalculatorImpl calculator = new CalculatorImpl();
        Expression expression = new Expression(Operator.Add, new Operand(2), new Operand(3));
        
        int result = calculator.calculate(expression);
        
        assertEquals(5, result);
    }
    
    @Test
    public void testSubtraction() {
        CalculatorImpl calculator = new CalculatorImpl();
        Expression expression = new Expression(Operator.Subtract, new Operand(5), new Operand(3));
        
        int result = calculator.calculate(expression);
        
        assertEquals(2, result);
    }
    
    @Test
    public void testMultiplication() {
        CalculatorImpl calculator = new CalculatorImpl();
        Expression expression = new Expression(Operator.Multiply, new Operand(4), new Operand(3));
        
        int result = calculator.calculate(expression);
        
        assertEquals(12, result);
    }
    
    @Test
    public void testDivision() {
        CalculatorImpl calculator = new CalculatorImpl();
        Expression expression = new Expression(Operator.Divide, new Operand(10), new Operand(2));
        
        int result = calculator.calculate(expression);
        
        assertEquals(5, result);
    }
}

File: src/test/kotlin/com/claymccoy/kcalc/CalculatorImplTest.kt
package com.claymccoy.kcalc

import kotlin.test.Test
import kotlin.test.assertEquals

class CalculatorImplTest {
    
    @Test
    fun testAddition() {
        val calculator = CalculatorImpl()
        val expression = Expression(Operator.Add, Operand(2), Operand(3))
        
        val result = calculator.calculate(expression)
        
        assertEquals(5, result)
    }
    
    @Test
    fun testSubtraction() {
        val calculator = CalculatorImpl()
        val expression = Expression(Operator.Subtract, Operand(5), Operand(3))
        
        val result = calculator.calculate(expression)
        
        assertEquals(2, result)
    }
    
    @Test
    fun testMultiplication() {
        val calculator = CalculatorImpl()
        val expression = Expression(Operator.Multiply, Operand(4), Operand(3))
        
        val result = calculator.calculate(expression)
        
        assertEquals(12, result)
    }
    
    @Test
    fun testDivision() {
        val calculator = CalculatorImpl()
        val expression = Expression(Operator.Divide, Operand(10), Operand(2))
        
        val result = calculator.calculate(expression)
        
        assertEquals(5, result)
    }
    
    @Test
    fun testExponent() {
        val calculator = CalculatorImpl()
        val expression = Expression(Operator.Exponent, Operand(2), Operand(3))
        
        val result = calculator.calculate(expression)
        
        assertEquals(8, result)
    }
}
